pandas/io/common.py

```diff
@@ -198,6 +198,8 @@ def validate_header_arg(header: object) -> None:
             raise ValueError("header must be integer or list of integers")
         if any(i < 0 for i in header):
             raise ValueError("cannot specify multi-index header with negative integers")
+        if list(header) != sorted(set(header)):
```

Comment:
Maybe `any(h1 >= h2 for h1, h2 in zip(header, header[:1])` so this check can short circuit

Reply:
The short circuit is a good idea. If header is already sorted, then `list(header) != sorted(set(header))` is either somewhat faster or comparable in speed to `any(h1 >= h2 for h1, h2 in zip(header, header[:1])`, but if header is not sorted than the short circuit makes a big difference. (I got curious and tested up to len(header) = 10**6.)